### PR TITLE
refactor(mito2): implement serialize/deserialize for RegionMetadata

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Implement `Serialize`/`Deserialize` directly on the struct instead of `Raw*`

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/mito2/src/metadata.rs [hunk 1]
```diff
@@ -56,21 +61,158 @@ pub(crate) struct RegionMetadata {
...
+
+        Ok(Self {
+            schema,
+            column_metadatas: region_metadata_without_schema.column_metadatas,
+            version: region_metadata_without_schema.version,
+            primary_keys: region_metadata_without_schema.primary_keys,
+            id: region_metadata_without_schema.id,
+        })
+    }
+}
```
[the remark this answers is not in the document]
Comment (waynexia) [reply src/mito2/src/metadata.rs:98]: Just want to get rid of `impl From<RegionMetadataWithoutSchema> for RegionMetadata`

[the remark this answers is not in the document]
Comment (evenyag) [reply src/mito2/src/metadata.rs:98]: Why?

File: src/mito2/src/metadata.rs [hunk 2]
```diff
@@ -56,21 +61,158 @@ pub(crate) struct RegionMetadata {
...
+
+    /// Add a column metadata to this region metadata.
+    /// This method will check the semantic type and add it to primary keys automatically.
+    pub fn add_column_metadata(mut self, column_metadata: ColumnMetadata) -> Self {
+        if column_metadata.semantic_type == SemanticType::Tag {
+            self.primary_keys.push(column_metadata.column_id);
+        }
+        self.column_metadatas.push(column_metadata);
+        self
+    }
```
[the remark this answers is not in the document]
Comment (waynexia) [reply src/mito2/src/metadata.rs:127]: I have yet to figure out how this builder will be used. Will take a look at that time

File: src/mito2/src/metadata.rs [hunk 3]
```diff
@@ -56,21 +61,140 @@ pub(crate) struct RegionMetadata {
     id: RegionId,
```
Comment (evenyag) [inline src/mito2/src/metadata.rs:61]: Could you also rename it to `region_id`? We use `region_id` in other places.

File: src/mito2/src/metadata.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (waynexia) [reply src/mito2/src/metadata.rs:98]: This is the internal detail of `fn deserialize()`. Implementing `From` may lead to unexpected usage.

**Review by evenyag — APPROVED** [review]

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**